homeassistant/components/renault/renault_entities.py

```diff
@@ -58,4 +58,8 @@ def extra_state_attributes(self) -> Mapping[str, Any] | None:
             last_update = self._get_data_attr("timestamp")
             if last_update:
                 return {ATTR_LAST_UPDATE: last_update}
+        if self.entity_description.coordinator == "location":
```

Comment:
Is `last_update` an isoformatted UTC datetime string?

This saves one line:
```py
@property
def extra_state_attributes(self) -> Mapping[str, Any] | None:
    """Return the state attributes of this entity."""
    last_update: str | None = None

    if self.entity_description.coordinator == "battery":
        last_update = self._get_data_attr("timestamp")
    elif self.entity_description.coordinator == "location":
        last_update = self._get_data_attr("lastUpdateTime")

    if last_update:
        return {ATTR_LAST_UPDATE: last_update}
    return None
```

Reply:
The format differs between various vehicles, so the renault API has kept it as a string.

Reply:
We should make sure that it's a UTC isoformatted timestamp though. That's our standard for times in state attributes.

Reply:
I have fixed the `mypy` issue.
However for the `last_update` I am not quite sure what you want me to do with the timestamps. These are sample timestamps I have in the fixtures:

1. `"timestamp": "2020-01-12T21:40:16Z"`
2. `"timestamp": "2020-11-17T09:06:48+01:00"`
3. `"lastUpdateTime": "2020-02-18T16:58:38Z"`

Are you saying that 1 and 3 are OK, but 2 should be converted to UTC?

Reply:
Looking at it through the UI, it seems to me that it handles both format so I think this is ok.

Reply:
No, all times in state attributes must be UTC time. Please convert the times to UTC.

Reply:
Is there a standard method in HA? or should I build my own?
```python
    def _convert_to_utc_string(self, value: str) -> str:
        """Return the state attributes of this entity."""
        original_dt = datetime.strptime(value, "%Y-%m-%dT%H:%M:%S%z")
        utc_dt = original_dt.astimezone(timezone.utc)
        return utc_dt.strftime("%Y-%m-%dT%H:%M:%S%z")
```

Reply:
Check the dt util. Maybe try starting with `parse_datetime`?

https://github.com/home-assistant/core/blob/46159c3f18bf3f405bced2f5280003e905748e1b/homeassistant/util/dt.py#L124

Reply:
I struggled a bit with pylint and mypy but should be ok now.